src/AudioUnit/AUParameter.cs

```diff
@@ -23,12 +23,16 @@ public string GetString (float? value)
 
 		public void SetValue (float value, AUParameterObserverToken originator)
 		{
+#pragma warning disable CS0618
 			SetValue (value, originator.ObserverToken);
+#pragma warning restore CS0618
```

Comment:
This needs review / comments, it hides something that should not happen.
IOW we should not depend on `[Obsolete]` API since we generally plan not to have them in `XAMCORE_4_0`. EIf a newer/better API exists then we should be using it (or we should comment why we can't)

Reply:
The obsolete method points to this method.

Reply:
> The obsolete method points to this method.

yes, that's exactly my point :)

Reply:
Fixing this properly would probably mean modifying the generator so we generate 2 overloads, one for AUParameterObserverToken and one for the IntPtr version (compat one).

How would one modify the generator so it generates _two_ methods and also forward `AUParameterObserverToken` to `AUParameterObserverToken.ObserverToken`

Reply:
Moving the non-disable warning fixes to a different PR.

Reply:
https://github.com/xamarin/xamarin-macios/pull/7642